### ISSUE TITLE
Zoom is limited for tall printer dimensions -- cannot zoom to fit

### ISSUE THREAD

**Issue by rtideas** [Ultimaker/Cura#3990]
**Application Version**
3.4

**Platform**
MacOS 10.13.5

**Printer**
Custom printer: 500(x)x500(y)x1800mm(z)

**Steps to Reproduce**
Create custom printer with 500 x 500 x 1800 mm dimensions.
![machine setup](https://user-images.githubusercontent.com/4334269/41995866-1fadfcf6-7a21-11e8-9fc5-6cfe1348af65.jpg)

**Actual Results**
Does not show full volume.

**Expected results**
Should be able to zoom out to show full volume--often called zoom to fit.

**Additional Information**
Prototyping a tall printer for a project. Created a machine setup that is 500mm(x) x 500mm(y) x 1800mm(z). I appears you are limited to har far out you can zoom.

Would assume you could zoom to fit.

[cone.3mf.zip](https://github.com/Ultimaker/Cura/files/2142687/cone.3mf.zip)

Standard 3d view
![3d_view](https://user-images.githubusercontent.com/4334269/41995823-ff639c12-7a20-11e8-9afe-393a38673e12.jpg)

3d view zoomed out to the max. Cannot see full object.
![3d_view_max_zoom_out](https://user-images.githubusercontent.com/4334269/41995822-ff559806-7a20-11e8-8de8-e20d93d97715.jpg)

front view
![front_view](https://user-images.githubusercontent.com/4334269/41995826-ff8ea970-7a20-11e8-975f-338c9fa716ab.jpg)

front view zoom out to max.
![front_view_max_zoom_out](https://user-images.githubusercontent.com/4334269/41995824-ff72b71a-7a20-11e8-8cf1-f5608cd17d75.jpg)

front view panned all the way to the top. Can only see a partial section of the object--never in full.
![front_view_panned_to_top](https://user-images.githubusercontent.com/4334269/41995825-ff81372c-7a20-11e8-9fba-a70c8aaf67bd.jpg)

top view panned -- Can see the object full, but not very useful.
![top_view_panned](https://user-images.githubusercontent.com/4334269/41995828-ffa64d78-7a20-11e8-8ccc-cd0813c34720.jpg)

**Comment by ChrisTerBeke** [comment]
You can pan by holding shift and then drag, or drag using the middle mouse button.

**Comment by fieldOfView** [comment]
OP obviously knows you can pan. Panning does not fix the problem.

**Comment by ChrisTerBeke** [comment]
I don't see what would solve the problem then, other than having a monitor with a different aspect ratio... I can zoom out fine to show the whole build volume (even when it's tall).

**Comment by fieldOfView** [comment]
I have an Ultimaker with Z Unlimited XXL which has a 1800 mm high build volume, and I can't get even close to seeing the whole buildvolume by zooming out and panning.

**Comment by rtideas** [comment]
Yes,the issue in not panning. There is a finite zoom level. The maximum zoom out value should be constricted by the machine volume.  

I would assume there is a value setting the minimum zoom level or visual scaling. Possibly it is derived from the maximum zoom?  

Changing to a theoretically 2000mm cube results in same limited zoom out ability. 

Without this ability, the ‘big picture’ cannot be seen.

**Comment by rtideas** [comment]
Would it have something to do with camera_distance?

https://github.com/Ultimaker/Cura/blob/master/resources/shaders/camera_distance.shader

**Comment by Ghostkeeper** [comment]
No, it's limited by this variable: https://github.com/Ultimaker/Uranium/blob/2ffdea15427b10eb58237f1e1928bf2cfa6bc9ee/plugins/Tools/CameraTool/CameraTool.py#L28

**Comment by Ghostkeeper** [comment]
It now sets the maximum zoom to some multiple of the length of the diagonal of the build volume.
